modin/experimental/core/execution/native/implementations/hdk_on_native/hdk_worker.py

```diff
@@ -56,13 +62,11 @@ def column_names(self) -> List[str]:
 
     @_inherit_docstrings(DbTable.to_arrow)
     def to_arrow(self) -> pa.Table:
-        at = getattr(self, "_at", None)
-        if at is None:
-            at = self._result.to_arrow()
-            if (names := getattr(self, "_column_names", None)) is not None:
-                at = at.rename_columns(names)
-            self._at = at
-        return at
+        return (
+            self._table.to_arrow()
+            if isinstance(self._table, ExecutionResult)
+            else self._table.run().to_arrow()
```

Comment:
This can cause performance degradation. Previously, you had zero cost for calling this method for imported arrow tables. Now, it would run actual query execution + arrow conversion for its result.

Reply:
Right, but it allows the arrow table to be garbage collected.
In case of a normal execution (no fallback to pandas), this method will not be called.